src/renderers/webgpu/utils/WebGPUTextureUtils.js

```diff
@@ -540,7 +540,7 @@ class WebGPUTextureUtils {
 
 		const readBuffer = device.createBuffer(
 			{
-				size: width * height * bytesPerTexel,
+				size: ( height - 1 ) * bytesPerRow + width * bytesPerTexel, // see https://github.com/mrdoob/three.js/issues/31658#issuecomment-3229442010
```

Comment:
Is this more readable?

```js
( width * bytesPerTexel ) + ( ( height - 1 ) * bytesPerRow )
```

Reply:
IMO, the "( height - 1 ) * bytesPerRow" bit should come first since "( width * bytesPerTexel )" represents the last row of data.

Reply:
Then this?

```js
( height - 1 ) * bytesPerRow + ( width * bytesPerTexel )
```

Or even this?

```js
( ( height - 1 ) * bytesPerRow ) + ( width * bytesPerTexel )
```